---
title: "Immunologic Landscape of MB"
output:
  html_notebook: default
  word_document: default
  html_document:
    df_print: paged
  pdf_document: default
---



```{r setup, include=FALSE,message=FALSE,warning=FALSE,echo=FALSE}
knitr::opts_chunk$set(
  comment = '', dev = c('pdf', 'tiff'),fig.width = 12,fig.height = 8)
library(tidyverse)
library(data.table)
library(pheatmap)
library(ComplexHeatmap)
library(rstatix)
library(circlize)
library(gprofiler2)
library(ggpubr)
library(circlize)
```

### Outline

- Fig.1 Mutation Burden
  - Fig. 1a All Variations, SNV+Indel+Fusionn
    - Fig. 1b SNV
    - Fig. 1c Indel
    - Fig. 1d Fusion
  - ~~Fig. 1e Top3 Frequent Mutations and Validated Mutations from literatures~~
- ~~Table 1 Count of Most Frequent/ Validated Mutations~~
- Fig. 2 Neoantigen Count
  - ~~Fig. 2a Neoantigen of MHC-I + II (optional, overlapped transcripts between I and II restricted)~~
    - Fig 2a MHC-I Neoantigens
      - Fig 2b MHC-I SNV/Indel/Fusion Neoantigens Breakdown
    - Fig. 2c  MHC-II Neoantigens
      - Fig. 2d MHC-I SNV/Indel/Fusion Neoantigens Breakdown
  - ~~Fig. 2e Neoantigen counts of Driver/Validated Mutations~~
  - Fig. S2 Fusions Visualization
- Table 1 Immunegenecity Rate of all patients neoantigens
- Table 2 Immunegenecity Rate of neoantigens by subgroups
- Table 3 Top3 Frequent Mutations and Validated Mutations pre/post pMHC
- Table 4a Most Frequent Mutations Print
- Table 4b Grp3/4 Mutations Print
- Fig. 3 Oncoprint
  - Fig. 3a Oncoprint of Frequent Mutations
  - Fig. 3b Oncoprint of Grp 3/4 Frequent Mutations
  - Fig. 3c Oncoprint of Grp 3/4 All Mutations
- Fig. 4 TAA
  -  Fig. 4a Protentially Targetable TAA
  -  ~~Fig. 4b pMHC of TAA~~
  - Fig. 4b pMHC I 
  - Fig. 4c pMHC II
  - Fig. 4d Heatmap of TAA/Fetal Cerebellum/Tesis/All Normal Tissues w pathway enrichment
  - Fig. s4 Interactive Pathway Enrichment of TAA / in html format
- Table 5 Immunegenecity Rate of all patients TAA
- Table 6 Immunegenecity Rate of TAA by subgroups
- Table 7 Top3 Frequent TAA pre/post pMHC
- Fig. 5 Concordance of Genomics and Proteomics 
- Fig. 6 MB Immunological Landscape (MA) 
 - Fig. 6a Antigen presentation/processing pathway
 - ~~ Fig. 6b MB Cibersortx scRNASeq Immune deconvolution~~
 - ~~Fig. 6c MB Inhibitory and Activating Ligand-Receptor Expression~~
 - ~~Fig. 6d B7H3 gene exression by all/subgroups~~


### Fig.1a Total Variations, SNV+Indel+MNV+Fusion

```{r message=FALSE,warning=FALSE}
mut <- fread("/blue/mitchell/changlin/MB_immune/Neoantigen/results/count/MB_ICGC_pre.tsv") %>% unique()
col_order <- c("WNT","SHH","Group 3","Group 4")
meta <- fread("/blue/mitchell/changlin/MB_immune/meta/northcott2017_RNAseq_phenotype_20190212.csv") %>% arrange(factor(SUBGROUP,levels=col_order)) %>% left_join(
survival <- fread("/blue/mitchell/changlin/MB_immune/meta/donor.PBCA-DE_survival_2019Nov.tsv"))
meta <- meta %>% arrange(factor(SUBGROUP,levels = col_order),factor(Donor_vital_status,levels = c("Alive","Deceased","NA")),desc(factor(Donor_survival_time)))


group_colors <- c("blue","red","gold","green4")
#data=data.frame(table(meta$SUBGROUP))
#my_xlab <- paste(levels(data$Var1),"\n(N=",table(meta$SUBGROUP),")",sep = "")
fusion_pre <- fread("/blue/mitchell/changlin/MB_immune/Neoantigen/results/fusions/MB_fusion_list.tsv")
fusion_pre <- fusion_pre %>% group_by(PID) %>% summarise(Transcript,N=n()) %>% ungroup() %>% dplyr::select(PID,N) %>% unique()

fig1a <- meta %>% left_join(mut) %>%left_join(fusion_pre) %>% mutate_all(funs(replace_na(.,0)))%>% mutate(total_moderat=SNV_moderat+Indel_moderat+MNV_moderat+N) 
fig1a$SUBGROUP <-factor(fig1a$SUBGROUP,levels = col_order)
fig1a_p <- fig1a %>% ggplot(aes(x=SUBGROUP,y=total_moderat))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "Total Count")

fig1a_wilcox <- fig1a %>% wilcox_test(total_moderat ~ SUBGROUP,p.adjust.method = "bonferroni",ref.group = "Group 3") %>% add_xy_position()
fig1a_p+stat_pvalue_manual(fig1a_wilcox,label = "p.adj.signif",tip.length = 0.01,xmin = "group1",xmax = "group2")+stat_compare_means(method = "kruskal.test",label.y = 210)



```

### Fig.1b Mutation Burden: SNV
```{r message=FALSE,warning=FALSE}

#Mutation VCF

fig1b <- meta %>% left_join(mut) 
fig1b$SUBGROUP <- factor(fig1b$SUBGROUP,levels = col_order)
fig1b_p <- fig1b %>% ggplot(aes(x=SUBGROUP,y=SNV_moderat))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "SNV")+ylim(0,100)

fig1b_wilcox <- fig1b %>% wilcox_test(SNV_moderat ~ SUBGROUP,p.adjust.method = "bonferroni",ref.group = "Group 3") %>% add_xy_position() %>% mutate(y.position=c(90,80,100))
fig1b_p+stat_pvalue_manual(fig1b_wilcox,label = "p.adj.signif",tip.length = 0.01,xmin = "group1",xmax = "group2")+stat_compare_means(method = "kruskal.test",label.y = 100)
```

### Fig.1c Mutation Burden: Indel
```{r message=FALSE,warning=FALSE}
fig1c <- meta %>% left_join(mut) 
fig1c$SUBGROUP <- factor(fig1c$SUBGROUP,levels = col_order)
fig1c_p <- fig1c %>% ggplot(aes(x=SUBGROUP,y=Indel_moderat))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "Indel")+ylim(0,5)
fig1c_wilcox <- fig1c %>% wilcox_test(Indel_moderat ~ SUBGROUP,p.adjust.method = "bonferroni",ref.group = "Group 3") %>% add_xy_position() %>% mutate(y.position=c(90,80,100))
fig1c_p+stat_pvalue_manual(fig1c_wilcox,label = "p.adj.signif",tip.length = 0.01,xmin = "group1",xmax = "group2")+stat_compare_means(method = "kruskal.test",label.y = 5)
```

### ~~Optional: Fig.1d Mutation Burden: MNV, There are no non-synonymous MNV. Won't show~~

```{r message=FALSE,warning=FALSE,fig.show=F, eval=FALSE}
 # meta %>% left_join(mut) %>% ggplot(aes(x=factor(meta$SUBGROUP,levels = col_order),y=MNV_moderat,color=SUBGROUP,fill=SUBGROUP))+
  # geom_violin(alpha=0.4)+
  # geom_jitter(height=0,width=0.2) +
  # scale_color_manual(values=group_colors)+
  # scale_fill_manual(values=group_colors)+
  #  theme(axis.line = element_line(size = 0.5, 
  #   linetype = "solid"), panel.background = element_rect(fill = NA, 
  #   linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "MNV")+ylim(0,1)
```


### Fig.1d Mutate Burden: Fusions
```{r message=FALSE,warning=FALSE}

fig1d <- meta %>% left_join(fusion_pre) %>% mutate_all(funs(replace_na(.,0)))
fig1d$SUBGROUP <- factor(fig1d$SUBGROUP,levels = col_order)
fig1d_p <- fig1d%>% ggplot(aes(x=SUBGROUP,y=N))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "Fusion")+ylim(0,5)
fig1d_p+stat_compare_means(method = "kruskal.test",label.y = 5)
```


### ~~Fig.1e Top3 Frequent Driver Mutations and Validated other MB Mutations (From Publications)~~

```{bash include=FALSE,warning=FALSE}
# awk '{print "MODERATE|"$1}' /blue/mitchell/changlin/MB_immune/Neoantigen/results/ICGC_Nov2019_hg38_vcf/Query_mutation_list.txt | fgrep -f - /blue/mitchell/changlin/MB_immune/Neoantigen/results/ICGC_Nov2019_hg38_vcf/ICGC_hg38_MB.vcf | sed "s/;/\t/g" | sed "s/|/\t/g" | awk '{print $8,$12}' | sed 's/_/\t/g' | awk '{print $1,$3}' OFS="\t" | sort | uniq | sed "1i\PID\tgene_name"> /blue/mitchell/changlin/MB_immune/Neoantigen/results/ICGC_Nov2019_hg38_vcf/validated_mutations_variation.txt -->

```


### ~~Table 1, Count of Most Frequenct Driver Mutations and Validated Mutations~~
```{r message=FALSE,warning=FALSE}
validated_mut <- fread("/blue/mitchell/changlin/MB_immune/Neoantigen/results/ICGC_Nov2019_hg38_vcf/validated_mutations_variation.txt") 
validated_mut <- validated_mut[validated_mut$PID %in% meta$PID,]
validated_mut <- validated_mut %>% left_join(meta) %>% dplyr::select(PID,gene_name,SUBGROUP)
#validated_mut %>% group_by(SUBGROUP,gene_name) %>% summarise(Mutation_Count=n()) %>% arrange(gene_name) %>% knitr::kable()

```

```{r message=FALSE,warning=FALSE}
mhc1_mut <- fread("/blue/mitchell/changlin/MB_immune/Neoantigen/results/count/MB_I.txt") %>% filter(source=="ICGC")
 Neo_I_SNV_Tx <- mhc1_mut   %>% filter(Variant_type=="missense",!duplicated(transcript_id),source=="ICGC") %>% group_by(PID)%>% summarise(Neo_I_SNV_Tx=n())  
  Neo_I_Indel_Tx <- mhc1_mut  %>% filter(Variant_type!="missense",!duplicated(transcript_id),source=="ICGC") %>% group_by(PID) %>% summarise(Neo_I_Indel_Tx=n())
Neo_I_Fusion_Tx <- fread("/blue/mitchell/changlin/MB_immune/Neoantigen/results/fusions/MB_fusion_pMHC_sf.tsv") %>% filter(HLA=="MHC_Class_I") %>% group_by(PID) %>% filter(!duplicated(Transcript_id)) %>% summarise(Neo_I_Fusion_Tx =n()) 
Neo_I_Tx <- Neo_I_Fusion_Tx  %>% full_join(Neo_I_SNV_Tx,by="PID")%>% full_join(Neo_I_Indel_Tx)%>%right_join(meta) %>% mutate_all(funs(replace_na(.,0))) %>% mutate(Neo_I_Tx=Neo_I_SNV_Tx+Neo_I_Indel_Tx+Neo_I_Fusion_Tx )
mhc2_mut <- fread("/blue/mitchell/changlin/MB_immune/Neoantigen/results/count/MB_II.txt")%>% filter(source=="ICGC")
 Neo_II_SNV_Tx <- mhc2_mut  %>% filter(Variant_type=="missense",!duplicated(transcript_id),source=="ICGC") %>% group_by(PID) %>% summarise(Neo_II_SNV_Tx=n())
  Neo_II_Indel_Tx <- mhc2_mut  %>% filter(Variant_type!="missense",!duplicated(transcript_id),source=="ICGC") %>% group_by(PID)%>% summarise(Neo_II_Indel_Tx=n())
Neo_II_Fusion_Tx <- fread("/blue/mitchell/changlin/MB_immune/Neoantigen/results/fusions/MB_fusion_pMHC_sf.tsv") %>% filter(HLA=="MHC_Class_II") %>% group_by(PID) %>% filter(!duplicated(Transcript_id)) %>% summarise(Neo_II_Fusion_Tx =n()) 
Neo_II_Tx <- Neo_II_Fusion_Tx  %>% full_join(Neo_II_SNV_Tx,by="PID")%>% full_join(Neo_II_Indel_Tx)%>%right_join(meta) %>% mutate_all(funs(replace_na(.,0))) %>% mutate(Neo_II_Tx=Neo_II_SNV_Tx+Neo_II_Indel_Tx+Neo_II_Fusion_Tx )
fusion_neo <- fread("/blue/mitchell/changlin/MB_immune/Neoantigen/results/fusions/MB_fusion_pMHC_sf.tsv") %>% separate(Gene,into = c("left_gene_name","right_gene_name"),sep = "_")
names(fusion_neo) <- str_replace(names(fusion_neo),"Transcript_id","transcript_id")

```


### ~~Fig. 2a Neoantigen MHC-I + MHC-II~~
*I and II may be overlapped*

```{r message=FALSE,warning=FALSE,fig.show=F}
 # Neo_I_Tx %>% left_join(Neo_II_Tx,by="PID") %>% transmute(PID,Neo_Tx=Neo_I_Tx+Neo_II_Tx) %>% left_join(meta)  %>% ggplot(aes(x=factor(SUBGROUP,levels = col_order),y=Neo_Tx,color=SUBGROUP,fill=SUBGROUP))+
 #  geom_violin(alpha=0.4)+
 #  geom_jitter(height=0,width=0.2) +
 #  scale_color_manual(values=group_colors)+
 #  scale_fill_manual(values=group_colors)+
 #   theme(axis.line = element_line(size = 0.5, 
 #    linetype = "solid"), panel.background = element_rect(fill = NA, 
 #    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "Neoantigen Count")+ylim(0,30)

 
```


### Fig.2a MHC-I Neoantigens

```{r message=FALSE,warning=FALSE}


## MHC-I Neoantigens
fig2a <- Neo_I_Tx 
fig2a$SUBGROUP <- factor(fig2a$SUBGROUP,levels=col_order)
fig2a_p <- fig2a%>% ggplot(aes(x=SUBGROUP,y=Neo_I_Tx))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "MHC-I Neoantigen")+ylim(0,20)
fig2a_wilcox <- fig2a %>% wilcox_test(Neo_I_Tx ~ SUBGROUP,p.adjust.method = "bonferroni",ref.group = "Group 3") %>% add_xy_position() %>% mutate(y.position=c(19,18,20))
fig2a_p+stat_pvalue_manual(fig2a_wilcox,label = "p.adj.signif",tip.length = 0.01,xmin = "group1",xmax = "group2")+stat_compare_means(method = "kruskal.test",label.y = 20)
```

### Fig.2b MHC-I SNV/Indel/Fusion Neoantigens

```{r message=FALSE,warning=FALSE}
theme_set(
  theme_classic()
)
fig2b <- Neo_I_Tx %>% dplyr::select(SUBGROUP,Neo_I_SNV_Tx,Neo_I_Indel_Tx,Neo_I_Fusion_Tx) %>%
  pivot_longer(!SUBGROUP,names_to="Alteration",values_to="Count") 
fig2b$SUBGROUP <-factor(fig2b$SUBGROUP,levels = col_order) 
fig2b$Alteration <- factor(fig2b$Alteration,levels=c("Neo_I_SNV_Tx","Neo_I_Indel_Tx","Neo_I_Fusion_Tx"))
fig2b_p <- fig2b %>% ggplot(aes(x=SUBGROUP,y=Count))+
  geom_boxplot(varwidth = TRUE,alpha=0.8,aes(fill=Alteration),outlier.alpha = 0.1)+
  ylim(0,10)+labs(x= "")
fig2b_p
```

### Fig.2c MHC-II Neoantigens
```{r message=FALSE,warning=FALSE}

fig2c <- Neo_II_Tx 
fig2c$SUBGROUP <- factor(fig2c$SUBGROUP,levels=col_order)
fig2c_p <- fig2c%>% ggplot(aes(x=SUBGROUP,y=Neo_II_Tx))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "MHC-II Neoantigen")+ylim(0,20)
fig2c_wilcox <- fig2c %>% wilcox_test(Neo_II_Tx ~ SUBGROUP,p.adjust.method = "bonferroni",ref.group = "Group 3") %>% add_xy_position() %>% mutate(y.position=c(19,18,20))
fig2c_p+stat_pvalue_manual(fig2c_wilcox,label = "p.adj.signif",tip.length = 0.01,xmin = "group1",xmax = "group2")+stat_compare_means(method = "kruskal.test",label.y = 20)
fig2c_summary <- fig2c %>% group_by(SUBGROUP) %>% get_summary_stats(Neo_II_Tx)

```

### Fig.2d MHC-II SNV/Indel/Fusion Neoantigens
```{r message=FALSE,warning=FALSE}
theme_set(
  theme_classic()
)
fig2d <- Neo_II_Tx %>% dplyr::select(SUBGROUP,Neo_II_SNV_Tx,Neo_II_Indel_Tx,Neo_II_Fusion_Tx) %>%
  pivot_longer(!SUBGROUP,names_to="Alteration",values_to="Count") 
fig2d$SUBGROUP <-factor(fig2d$SUBGROUP,levels = col_order) 
fig2d$Alteration <- factor(fig2d$Alteration,levels=c("Neo_II_SNV_Tx","Neo_II_Indel_Tx","Neo_II_Fusion_Tx"))
fig2d_p <- fig2d %>% ggplot(aes(x=SUBGROUP,y=Count))+
  geom_boxplot(varwidth = TRUE,alpha=0.8,aes(fill=Alteration),outlier.alpha = 0.1)+
  ylim(0,10)+labs(x= "")
fig2d_p
```


## Fig. Supp 2 Medulloblastoma Fusion Visulization

```{r message=FALSE,warning=FALSE}
# load("/blue/mitchell/changlin/MB_immune/Nooantigen/scripts/Fusion_vis.RData")
# tiff("real_loc_I.tiff",width = 1600,height = 1600,res = 300)
# circos.initializeWithIdeogram()
# text(0, 0,"MB Fusion MHC_I", cex = 1)
# circos.genomicLink(bed_left_I,bed_right_I,col=fusion_pMHC_I$color,lwd =fusion_pMHC_I$count,lty=1)
# dev.off()
# 
# tiff("real_loc_II.tiff",width = 1600,height = 1600,res = 300)
# circos.initializeWithIdeogram()
# text(0, 0,"MB Fusion MHC_II", cex = 1)
# circos.genomicLink(bed_left_II,bed_right_II,col=fusion_pMHC_II$color,lwd =fusion_pMHC_II$count,lty=1)
# dev.off()
# 
# tiff("real_loc_pre.tiff",width = 1600,height = 1600,res = 300)
# circos.initializeWithIdeogram()
# text(0, 0,"MB Fusion Proteins", cex = 1)
# circos.genomicLink(bed_pre_left,bed_pre_right,col=fusion_pre$color,lwd =fusion_pre$count,lty=1)
# dev.off()
```


## Fig. S2a Protentially targetable Fusions by by Molecular Subgroups

![Fig. S2a Protentially targetable Fusions by subgroups](./real_loc_pre.tiff)

## Fig. S2b MHC-I restricted Immunogenic Fusions by Molecular Subgroups

![Fig. S2b MHC-I restricted Immunogenic Fusions by Molecular Subgroups](./real_loc_I.tiff)


## Fig. S2c MHC-II restricted Immunogenic Fusions by Molecular Subgroups

![Fig. S2c MHC-II restricted Immunogenic Fusions by Molecular Subgroups](./real_loc_II.tiff)




### Table 1a What percent of patients have at least 1 targetable neoantigen?

```{r message=FALSE,warning=FALSE}

total_freq <- rbind(mhc1_mut,mhc2_mut) %>% dplyr::select(PID,transcript_id) %>% rbind(fusion_neo%>% dplyr::select(PID,transcript_id)) %>%
  unique() %>% group_by(PID) %>% summarise(N=n()) %>% full_join(meta)

table1a <- tibble(`One+`=paste0(round((total_freq %>% filter(N>0) %>%nrow() /170)*100,1),'%'),
           `Two+`=paste0(round((total_freq %>% filter(N>1) %>%nrow() /170)*100,1),'%'),
           `Three+`=paste0(round((total_freq %>% filter(N>2) %>%nrow() /170)*100,1),'%'))
table1a %>% knitr::kable()
```

### Table 1b What percent of patients (by subgroup) have at least one targetable neoantigen?

```{r message=FALSE,warning=FALSE}
table1b <- total_freq %>% filter(N>0) %>% group_by(SUBGROUP) %>%summarise(N=n()) %>% cbind(table(meta$SUBGROUP) %>% as.data.frame())%>% transmute(SUBGROUP,`One+`=paste0(round(N/Freq*100,1),'%')) %>%left_join(total_freq %>% filter(N>1) %>% group_by(SUBGROUP) %>%summarise(N=n()) %>% cbind(table(meta$SUBGROUP) %>% as.data.frame())%>% transmute(SUBGROUP,`Two+`=paste0(round(N/Freq*100,1),'%'))) %>% left_join(total_freq %>% filter(N>2) %>% group_by(SUBGROUP) %>%summarise(N=n()) %>% cbind(table(meta$SUBGROUP) %>% as.data.frame())%>% transmute(SUBGROUP,`Three+`=paste0(round(N/Freq*100,1),'%'))) %>% arrange(factor(SUBGROUP,levels = col_order))
table1b %>% knitr::kable()


```

### Table 2 Immungencity of validated mutations
```{r message=FALSE,warning=FALSE}
val_neo_mut <- mhc1_mut %>% rbind(mhc2_mut) %>% filter(source=="ICGC") 
table2 <- validated_mut %>% group_by(SUBGROUP,gene_name) %>% summarise(Mutation_Count=n()) %>% arrange(gene_name) %>%
 left_join((val_neo_mut[val_neo_mut$gene_name %in% validated_mut$gene_name,] %>% left_join(meta) %>% dplyr::select(PID,SUBGROUP,gene_name) %>% unique() %>%  group_by(SUBGROUP,gene_name) %>% summarise(Neoantigen_Count=n()) %>%arrange(gene_name)),by=c("SUBGROUP","gene_name")) %>% mutate_all(funs(replace_na(.,0)))
table2 %>% knitr::kable()

```

### Table 3a Recurrent Candidate Neoantigens, span at least 3 patients (showing first 10 lines)

```{r message=FALSE}
mhc1_mut$HLA_type="HLA-I" 
mhc2_mut$HLA_type="HLA-II"

mhc1_mut <- mhc1_mut %>% dplyr::select(PID,gene_name,Variant_type,HLA_type)
mhc2_mut <- mhc2_mut %>% dplyr::select(PID,gene_name,Variant_type,HLA_type)
All_mut <- mhc1_mut %>% full_join(mhc2_mut,by=c("PID","gene_name","Variant_type")) %>% unique()
All_mut <- All_mut %>% mutate(HLA_I_SNV=ifelse(Variant_type=="missense" & HLA_type.x=="HLA-I","HLA_I_SNV",""),HLA_I_Indel=ifelse(Variant_type !="missense" & HLA_type.x=="HLA-I","HLA_I_Indel",""),HLA_II_SNV=ifelse(Variant_type=="missense" & HLA_type.y=="HLA-II","HLA_II_SNV",""),HLA_II_Indel=ifelse(Variant_type !="missense" & HLA_type.y=="HLA-II","HLA_II_Indel","")) %>%  dplyr::select(PID,gene_name,contains("HLA_I")) %>% left_join(meta)
 All_mut[is.na(All_mut)]=""

# N is out of total not out of each subgroup hereby
All_mut <- All_mut %>% mutate(ICGC_alt=paste(HLA_I_SNV,HLA_I_Indel,HLA_II_SNV,HLA_II_Indel,sep = ";")) %>% dplyr::select(PID,SUBGROUP,ICGC_alt,gene_name)

All_mut_freq <- All_mut %>% group_by(gene_name) %>% summarise(PID,SUBGROUP,ICGC_alt,N=n()) %>% filter(N>2) 

MB_ICGC <- All_mut %>% dplyr::select(!SUBGROUP)%>%pivot_wider(names_from = 'PID',values_from='ICGC_alt') %>% column_to_rownames(var = "gene_name")
MB_ICGC[is.na(MB_ICGC)]=""
meta_neo <- meta[meta$PID %in% colnames(MB_ICGC),]
MB_ICGC_freq <- MB_ICGC[rownames(MB_ICGC) %in% All_mut_freq$gene_name,] %>% dplyr::select(meta_neo$PID)

#Table 3a
table3a <- All_mut_freq %>% arrange(desc(N))
table3a %>%head(10)%>% knitr::kable()
```

### Table 3b Group3/4 Neoantigens (showing first 10 lines)

```{r message=FALSE}
meta_grp3_4 <- meta %>% filter(str_detect(SUBGROUP,"Group"))
grp3_4_mut <- All_mut[All_mut$PID %in% meta_grp3_4$PID,] 
grp3_4_neo <- MB_ICGC %>% dplyr::select(grp3_4_mut$PID)
meta_grp3_4 <- meta_grp3_4[meta_grp3_4$PID %in% grp3_4_mut$PID]

grp3_4_1_mut <- grp3_4_mut %>% group_by(gene_name) %>% summarise(PID,SUBGROUP,ICGC_alt,N=n())%>%arrange(desc(N)) %>% filter(N>0)
grp3_4_2_mut <- grp3_4_mut %>% group_by(gene_name) %>% summarise(PID,SUBGROUP,ICGC_alt,N=n())%>%arrange(desc(N)) %>%filter(N>1)
grp3_4_neo_1 <- grp3_4_neo[rownames(grp3_4_neo) %in% grp3_4_1_mut$gene_name,] %>% dplyr::select(meta_grp3_4$PID)
grp3_4_neo_2 <- grp3_4_neo[rownames(grp3_4_neo) %in% grp3_4_2_mut$gene_name,]%>% dplyr::select(meta_grp3_4$PID)



#Table 3b
table3b <- grp3_4_1_mut
table3b %>% head(10)%>% knitr::kable() 
```

### Fig. 3a-3c Oncoprint of Recurrent Neoantigens and Group3/4 Recurrent/All Neoantigens
Oncoprint
```{r message=F,warning=F}
col_HLA = c("HLA_I_SNV" = "blue", "HLA_I_Indel" = "orange", "HLA_II_SNV" = "red","HLA_II_Indel"="green")

alter_fun = list(
  background = function(x, y, w, h) {
    grid.rect(x, y, w-unit(4, "pt"), h-unit(2, "pt"), 
              gp = gpar(fill = "#CCCCCC", col = NA))
  },
  # big blue
  HLA_I_SNV = function(x, y, w, h) {
    grid.rect(x, y, w-unit(4, "pt"), h-unit(2, "pt"), 
              gp = gpar(fill = col_HLA["HLA_I_SNV"], col = NA))
  },
  # big red
  HLA_I_Indel = function(x, y, w, h) {
    grid.rect(x, y, w-unit(4, "pt"), h-unit(2, "pt"), 
              gp = gpar(fill = col_HLA["HLA_I_Indel"], col = NA))
  },
  # small orange
  HLA_II_SNV = function(x, y, w, h) {
    grid.rect(x, y, w-unit(2, "pt"), h*0.33, 
              gp = gpar(fill = col_HLA["HLA_II_SNV"], col = NA))
  },
  #small green
  HLA_II_Indel = function(x, y, w, h) {
    grid.rect(x, y, w-unit(4, "pt"), h*0.33, 
              gp = gpar(fill = col_HLA["HLA_II_Indel"], col = NA))
  }
)

column_title_ICGC = "ICGC Medulloblastoma Neoantigens (n=170)"
column_title_grp3_4 = "ICGC Group3/4 Medulloblastoma Neoantigens (n=71)"
heatmap_legend_param = list(title = "Alternations", at = c("HLA_I_SNV","HLA_I_Indel","HLA_II_SNV","HLA_II_Indel"), 
                            labels = c("HLA_I_SNV","HLA_I_Indel","HLA_II_SNV","HLA_II_Indel"))
# ICGC All patients, Variantions span 3 or more patients
ICGC_all <- oncoPrint(MB_ICGC_freq,
          alter_fun = alter_fun, col = col_HLA, 
          column_order = meta_neo$PID,
          heatmap_legend_param=heatmap_legend_param,
          column_title = column_title_ICGC,
          top_annotation = HeatmapAnnotation(Subtype=meta_neo$SUBGROUP,
                                             Age=meta_neo$AGE,
                                             Gender=meta_neo$GENDER,
                                             Vital_Status=meta_neo$Donor_vital_status,
                                             Overall_Survival=anno_lines(meta_neo$Donor_survival_time,gp=gpar(frontsize=8)),
                                               which = "col",
                                             col=list(Subtype=c("WNT"="blue","SHH"="red","Group 3"="yellow","Group 4"="green"),Gender=c("Female"="red","Male"="blue","Tuner"="brown","Klinefelter"="green"),Vital_Status=c("Alive"="red","Deceased"="blue"))),
)

# At least 2 neoantigens in Grp3/4
ICGC_grp3_4_2 <- oncoPrint(grp3_4_neo_2,
          alter_fun = alter_fun, col = col_HLA, 
          column_order = meta_grp3_4$PID,
          heatmap_legend_param=heatmap_legend_param,
          column_title = column_title_grp3_4,
           top_annotation = HeatmapAnnotation(Subtype=meta_grp3_4$SUBGROUP,
                                             Age=meta_grp3_4$AGE,
                                             Gender=meta_grp3_4$GENDER,
                                             Vital_Status=meta_grp3_4$Donor_vital_status,
                                             Overall_Survival=anno_lines(meta_grp3_4$Donor_survival_time,gp=gpar(frontsize=8)),
                                               which = "col",
                                             col=list(Subtype=c("WNT"="blue","SHH"="red","Group 3"="yellow","Group 4"="green"),Gender=c("Female"="red","Male"="blue","Tuner"="brown","Klinefelter"="green"),Vital_Status=c("Alive"="red","Deceased"="blue"))),
)
#At least 1 antigen in Grp3/4
ICGC_grp3_4_1<- oncoPrint(grp3_4_neo_1,
          alter_fun = alter_fun, col = col_HLA, 
          column_order = meta_grp3_4$PID,
          row_names_gp = gpar(fontsize=4),
         show_row_names = F,
         #remove_empty_rows = T,
          show_pct = F,
          heatmap_legend_param=heatmap_legend_param,
          column_title = column_title_grp3_4,
          top_annotation = HeatmapAnnotation(Subtype=meta_grp3_4$SUBGROUP,
                                             Age=meta_grp3_4$AGE,
                                             Gender=meta_grp3_4$GENDER,
                                             Vital_Status=meta_grp3_4$Donor_vital_status,
                                             Overall_Survival=anno_lines(meta_grp3_4$Donor_survival_time,gp=gpar(frontsize=8)),
                                               which = "col",
                                             col=list(Subtype=c("WNT"="blue","SHH"="red","Group 3"="yellow","Group 4"="green"),Gender=c("Female"="red","Male"="blue","Tuner"="brown","Klinefelter"="green"),Vital_Status=c("Alive"="red","Deceased"="blue"))),
)
draw(ICGC_all,annotation_legend_side="bottom",heatmap_legend_side="bottom",merge_legend=T)
draw(ICGC_grp3_4_2,annotation_legend_side="bottom",heatmap_legend_side="bottom",merge_legend=T)
draw(ICGC_grp3_4_1,annotation_legend_side="bottom",heatmap_legend_side="bottom",merge_legend=T)
```

### Fig. 4a Protentially Targetable TAA
```{r message=F,warning=F}
TAA <- fread("/blue/mitchell/changlin/MB_immune/TAA/results/TAA_fetal_testis.txt")
files <- list.files(path="/blue/mitchell/changlin/MB_immune/TAA/results", pattern = "*_u_2sd.tsv",all.files=T,full.names = T)
TAA_count <- data.frame(matrix(nrow = 1,ncol = 0))
  for(file in files){
  file_id <- fread(file,header = T)
  file_name <- basename(file) %>% str_remove(".isoforms.results_u_2sd.tsv")
  TAA_count$file_name <- nrow(file_id)
  names(TAA_count) <- names(TAA_count) %>% str_replace_all("file_name",file_name) 
}
TAA_count <-t(TAA_count) %>% as.data.frame() %>% rownames_to_column(var="PID")
names(TAA_count) <- names(TAA_count) %>% str_replace_all("V1","Count")
fig4a <- meta %>% left_join(TAA_count) 
fig4a$SUBGROUP <- factor(fig4a$SUBGROUP,levels = col_order)
fig4a_p <- fig4a %>% ggplot(aes(x=SUBGROUP,y=Count))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "TAA Count")+ylim(0,50)

fig4a_wilcox <- fig4a %>% wilcox_test(Count ~ SUBGROUP,p.adjust.method = "bonferroni",ref.group = "Group 3") %>% add_xy_position() %>% mutate(y.position=c(46,43,48))
fig4a_p+stat_pvalue_manual(fig4a_wilcox,label = "p.adj.signif",tip.length = 0.01,xmin = "group1",xmax = "group2")+stat_compare_means(method = "kruskal.test",label.y = 50)



#TAA pMHC, I 8-15 AA,II 15AA
TAA_I_count <- fread("/blue/mitchell/changlin/MB_immune/TAA/results/count/TAA_HLA_I.txt")
TAA_II_count <- fread("/blue/mitchell/changlin/MB_immune/TAA/results/count/TAA_HLA_II.txt")
TAA_all_count <- TAA_I_count %>% full_join(TAA_II_count) %>% full_join(meta) %>% mutate_all(funs(replace_na(.,0))) %>% mutate(Count=HLA_I_Count+HLA_II_Count)
```

### ~~Fig. 4b~~
*TAA pMHC I+II plot, I and II may be overlapped*
```{r message=F,eval=FALSE,fig.show=FALSE}
# TAA_all_count %>% ggplot(aes(x=factor(TAA_all_count$SUBGROUP,levels = col_order),y=Count,color=SUBGROUP,fill=SUBGROUP))+
#   geom_violin(alpha=0.4)+
#   geom_jitter(height=0,width=0.2) +
#   scale_color_manual(values=group_colors)+
#   scale_fill_manual(values=group_colors)+
#    theme(axis.line = element_line(size = 0.5, 
#     linetype = "solid"), panel.background = element_rect(fill = NA, 
#     linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "TAA Count")+ylim(0,50)

```

### Fig. 4b-c TAA Immunogenic Transcripts by MHC Types
```{r message=F}
# TAA_all_count %>% dplyr::select(SUBGROUP,HLA_I_Count,HLA_II_Count) %>%
#   pivot_longer(!SUBGROUP,names_to="HLA_Type",values_to="Count") %>% ggplot(aes(x=factor(SUBGROUP,levels = col_order),y=Count,fill=HLA_Type ))+geom_boxplot(varwidth = TRUE,alpha=0.8)+scale_y_log10(oob=scales::squish_infinite)+labs(x= "")

fig4b <- TAA_I_count %>% left_join(meta)
fig4b$SUBGROUP <- factor(fig4b$SUBGROUP,levels = col_order)
fig4b_p <- fig4b %>% ggplot(aes(x=SUBGROUP,y=HLA_I_Count))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "TAA MHC-I Count")+ylim(0,40)

fig4b_wilcox <- fig4b %>% wilcox_test(HLA_I_Count ~ SUBGROUP,p.adjust.method = "bonferroni",ref.group = "Group 3") %>% add_xy_position() %>% mutate(y.position=c(38,36,34))
fig4b_p+stat_pvalue_manual(fig4b_wilcox,label = "p.adj.signif",tip.length = 0.01,xmin = "group1",xmax = "group2")+stat_compare_means(method = "kruskal.test",label.y = 40)

fig4c <- TAA_II_count %>% left_join(meta)
fig4c$SUBGROUP <- factor(fig4c$SUBGROUP,levels = col_order)
fig4c_p <- fig4c %>% ggplot(aes(x=SUBGROUP,y=HLA_II_Count))+
  geom_violin(alpha=0.4,aes(fill=SUBGROUP))+
  geom_jitter(height=0,width=0.2) +
  scale_color_manual(values=group_colors)+
  scale_fill_manual(values=group_colors)+
   theme(axis.line = element_line(size = 0.5, 
    linetype = "solid"), panel.background = element_rect(fill = NA, 
    linetype = "solid"), plot.background = element_rect(linetype = "solid")) +labs(x= "",y = "TAA MHC-II Count")+ylim(0,40)

fig4c_wilcox <- fig4c %>% wilcox_test(HLA_II_Count ~ SUBGROUP,p.adjust.method = "bonferroni",ref.group = "Group 3") %>% add_xy_position() %>% mutate(y.position=c(38,36,34))
fig4c_p+stat_pvalue_manual(fig4c_wilcox,label = "p.adj.signif",tip.length = 0.01,xmin = "group1",xmax = "group2")+stat_compare_means(method = "kruskal.test",label.y = 40)



```

### Fig. 4d Unsupervised Hierarchical Cluster Analysis of Medulloblastoma Tumor Associated Antigens
C1 = Oncofetal Antigens
C2 = Low Confidence Cancer Testis Antigens (TPM 1e0-1e1) and others
C3 = High Confidence Cancer Testis Antigens (TPM 1e1-1e2)
```{r message=F}
mb_fetal_testis_normal <- fread("/blue/mitchell/changlin/MB_immune/TAA/results/TAA_fetal_testis_normal_tx.txt") %>% column_to_rownames(var="transcript_id") %>% dplyr::select(!contains("Adult"))
mb_fetal_testis_normal <- mb_fetal_testis_normal %>% dplyr::select(meta$PID,everything())
mb_alone_taa <- mb_fetal_testis_normal[,c(1:170)]
col_name_pheatmap <- fread("/blue/mitchell/changlin/MB_immune/TAA/results/TAA_fetal_testis_normal_tx_name.txt") %>% arrange(factor(PID,levels = colnames(mb_fetal_testis_normal))) %>% column_to_rownames(var="PID") 
 col_name_complex <- fread("/blue/mitchell/changlin/MB_immune/TAA/results/TAA_fetal_testis_normal_tx_name.txt") %>% arrange(factor(PID,levels = colnames(mb_fetal_testis_normal)))
 col_name_complex$Group <- factor(col_name_complex$Group,levels = c("WNT","SHH","Group 3","Group 4","Fetal Cerebellum","Testis","Normal"))
 taa_all_enst <- fread("/blue/mitchell/changlin/MB_immune/TAA/results/cTAA_dict/pid_enst_pMHC.txt") %>% left_join(meta)
## TAA pre

# 
# f1=colorRamp2(c(0,0.5,1),c("#4575B4","#FEFEC0","#D73027"))
# set.seed(1234)
# kclus <- kmeans(mb_fetal_testis_normal,3,iter=5000)
# split <- paste0("Cluster\n",kclus$cluster)
# cluster_scale <- data.frame(Cluster=sort(kclus$cluster))
# cluster_scale$transcript_id <- row.names(cluster_scale)
# cluster_scale <- cluster_scale %>% left_join(TAA) %>% mutate(testis_max=2^testis_max)
# 
# hm_TAA=Heatmap(as.matrix(mb_fetal_testis_normal),
#                use_raster = T,raster_device = "tiff",
#                col = f1,
#                row_split = split,
#                row_dend_reorder = TRUE,
#                cluster_columns = F,
#               cluster_rows = T,
#                show_row_names = F,
#                show_column_names = F,
#                name="Gene Expression",
#                column_title = "ICGC MB TAA",
#                column_title_gp = gpar(fontsize=16,fontface="bold"),
#                row_title = "TPM",
#                row_title_gp = gpar(fontsize=16,fontface="bold"),
#                border=T,
#                clustering_distance_rows = "euclidean",
#                clustering_method_rows = "complete",
#                heatmap_legend_param = list(
#                  title = "TPM",
#                  title_position = "lefttop-rot",labels_gp=gpar(fontsize=12),title_gp=gpar(fontsize=12)),
#                top_annotation = HeatmapAnnotation(Group=col_name_complex$Group,
#                                                    which = "col",
#                                                   col=list(Group=c("WNT"="blue","SHH"="red","Group 3"="yellow","Group 4"="green","Fetal"="orange","Testis"="gray","Normal"="black"),Gender=c("Female"="red","Male"="blue","Tuner"="brown","Klinefelter"="green")),
#                                                   annotation_name_offset=unit(0.5,"mm"),show_annotation_name = T))
# 
# old.pos <- data.frame(transcript_id=rownames(mb_fetal_testis_normal),old_pos=c(1:nrow(mb_fetal_testis_normal)))
# new.pos <- data.frame(old_pos=unlist(row_order(hm_TAA)),new_pos=c(1:nrow(mb_fetal_testis_normal))) %>% right_join(old.pos) %>% right_join(cluster_scale) %>% arrange(new_pos)
# taa_no_antigen <- tibble(transcript_id=TAA$transcript_id[!TAA$transcript_id %in% taa_all_enst$transcript_id] ) %>% left_join(new.pos)

# TAA post pMHC



f1=colorRamp2(c(0,0.5,1,10),c("#4575B4","#FEFEC0","red","red4"))
set.seed(1234)
mb_fetal_testis_normal_post <- mb_fetal_testis_normal[rownames(mb_fetal_testis_normal) %in% taa_all_enst$transcript_id,]

kclus <- kmeans(mb_fetal_testis_normal_post,3,iter=5000)
split <- paste0("Cluster\n",kclus$cluster)
cluster_scale <- data.frame(Cluster=sort(kclus$cluster))
cluster_scale$transcript_id <- row.names(cluster_scale)
cluster_scale <- cluster_scale %>% left_join(TAA) %>% mutate(testis_max=2^testis_max)
left_annotation <- rowAnnotation(block = anno_block(gp = gpar(2:4),
                                                      labels = c("C1", "C2", "C3"), labels_gp = gpar(col = c("black", "blue", "orange"),fontsize=9)))
hm_TAA_post=Heatmap(as.matrix(mb_fetal_testis_normal_post),
               use_raster = T,raster_device = "tiff",
               col = f1,
               row_split = split,
               row_dend_reorder = TRUE,
               cluster_columns = F,
              cluster_rows = T,
               show_row_names = F,
               show_column_names = F,
               name="Gene Expression",
               column_title = "ICGC MB TAA",
               column_title_gp = gpar(fontsize=16,fontface="bold"),
               row_title = "TPM",
               row_title_gp = gpar(fontsize=16,fontface="bold"),
               border=T,
               clustering_distance_rows = "euclidean",
               clustering_method_rows = "complete",
               heatmap_legend_param = list(
                 title = "TPM",
                 title_position = "lefttop-rot",labels_gp=gpar(fontsize=12),title_gp=gpar(fontsize=12)),
              left_annotation = left_annotation,
               top_annotation = HeatmapAnnotation(Group=col_name_complex$Group,
                                                   which = "col",
                                                  col=list(Group=c("WNT"="blue","SHH"="red","Group 3"="yellow","Group 4"="green","Fetal Cerebellum"="orange","Testis"="gray","Normal"="black"),Gender=c("Female"="red","Male"="blue","Tuner"="brown","Klinefelter"="green")),
                                                  annotation_name_offset=unit(0.5,"mm"),show_annotation_name = T))

old.pos <- data.frame(transcript_id=rownames(mb_fetal_testis_normal_post),old_pos=c(1:nrow(mb_fetal_testis_normal_post)))
new.pos <- data.frame(old_pos=unlist(row_order(hm_TAA_post)),new_pos=c(1:nrow(mb_fetal_testis_normal_post))) %>% right_join(old.pos) %>% right_join(cluster_scale) %>% arrange(new_pos)

hm_TAA_post
draw(hm_TAA_post,annotation_legend_side="bottom",heatmap_legend_side="right",merge_legend=T,newpage=FALSE)


```

### Fig. Supp 4 Pathway Enrichment of TAA
```{r message=F}
gostres_all <- gost(query =c(cluster_scale$gene_name),organism=("hsapiens"))
p <- gostplot(gostres_all,capped = FALSE,interactive = TRUE)


# gostres_1 <- gost(query =c((cluster_scale %>% filter(Cluster=="1"))$gene_name),organism=("hsapiens"))
# gostres_3 <- gost(query =c((cluster_scale %>% filter(Cluster=="3"))$gene_name),organism=("hsapiens"))
# gostres_2 <- gost(query =c((cluster_scale %>% filter(Cluster=="2"))$gene_name),organism=("hsapiens"))
# gostres_4 <- gost(query =c((cluster_scale %>% filter(Cluster=="4"))$gene_name),organism=("hsapiens"))

# p1 <- gostplot(gostres_1,capped = FALSE,interactive = TRUE)
# p4 <- gostplot(gostres_4,capped = FALSE,interactive = TRUE)

# head(gostres_all$result)
p

```
### Table 5 Immunegenecity Rate of all patients TAA
```{r}

total_freq_taa <- taa_all_enst %>% dplyr::select(PID,transcript_id) %>%
  unique() %>% group_by(PID) %>% summarise(N=n()) %>% full_join(meta)

table5 <- tibble(`One+`=paste0(round((total_freq_taa %>% filter(N>0) %>%nrow() /170)*100,1),'%'),
           `Two+`=paste0(round((total_freq_taa %>% filter(N>1) %>%nrow() /170)*100,1),'%'),
           `Three+`=paste0(round((total_freq_taa %>% filter(N>2) %>%nrow() /170)*100,1),'%'))
table5 %>% knitr::kable()
```

### Table 6 Immunegenecity Rate of TAA by subgroups
```{r}
table6 <- total_freq_taa %>% filter(N>0) %>% group_by(SUBGROUP) %>%summarise(N=n()) %>% cbind(table(meta$SUBGROUP) %>% as.data.frame())%>% transmute(SUBGROUP,`One+`=paste0(round(N/Freq*100,1),'%')) %>%left_join(total_freq %>% filter(N>1) %>% group_by(SUBGROUP) %>%summarise(N=n()) %>% cbind(table(meta$SUBGROUP) %>% as.data.frame())%>% transmute(SUBGROUP,`Two+`=paste0(round(N/Freq*100,1),'%'))) %>% left_join(total_freq %>% filter(N>2) %>% group_by(SUBGROUP) %>%summarise(N=n()) %>% cbind(table(meta$SUBGROUP) %>% as.data.frame())%>% transmute(SUBGROUP,`Three+`=paste0(round(N/Freq*100,1),'%'))) %>% arrange(factor(SUBGROUP,levels = col_order))
table6 %>% knitr::kable()
```

### Table 7 Top3 Frequent TAA pre/post pMHC

```{r message=F,warning=F}
mb_taa_pre <- mb_alone_taa %>% transmute(expressed_count=apply(mb_alone_taa,1,function(x)sum(x>1))) %>% arrange(desc(expressed_count)) %>% rownames_to_column(var="transcript_id") 
mb_taa_pre_top5 <- mb_taa_pre[c(1:5),] %>% left_join(TAA) %>% dplyr::select(transcript_id,gene_name,expressed_count)
mb_taa_pre_top5_count <- mb_alone_taa %>% rownames_to_column(var="transcript_id")
mb_taa_pre_top5_count <- mb_taa_pre_top5_count[mb_taa_pre_top5_count$transcript_id %in%mb_taa_pre_top5$transcript_id, ] %>% pivot_longer(!transcript_id,names_to="PID",values_to="TPM") %>% filter(TPM>1) %>% left_join(meta)
mb_taa_post_top5_count <- taa_all_enst[taa_all_enst$transcript_id %in% mb_taa_pre_top5$transcript_id,]

table7 <- mb_taa_pre_top5_count %>% group_by(SUBGROUP,transcript_id) %>% summarise(TAA_Pre_pMHC_Count=n()) %>% 
 left_join(mb_taa_post_top5_count %>% dplyr::select(PID,SUBGROUP,transcript_id) %>% unique() %>%  group_by(SUBGROUP,transcript_id) %>% summarise(TAA_Post_pMHC_Count=n()),by=c("SUBGROUP","transcript_id")) %>% mutate_all(funs(replace_na(.,0))) %>% left_join(TAA) %>% dplyr::select(SUBGROUP,gene_name,TAA_Pre_pMHC_Count,TAA_Post_pMHC_Count) %>% arrange(gene_name)
table7 %>% knitr::kable()

```



### Fig. 5a MB Proteomics vs Genomics
```{r message=F}
fig5a <- fread("/orange/mitchell/MB_immune/Mass/MB_mass_count.txt") %>% left_join(meta)%>% arrange(factor(SUBGROUP,levels=col_order))%>% mutate(pid=paste(SUBGROUP,PID,sep = "_")) %>% dplyr::select(pid,c(2:9))%>% pivot_longer(!pid,names_to="Type",values_to="Count") %>% separate(Type,sep = "_",into = c("Type","Platform")) %>% mutate_all(funs(str_replace_all(.,c("Tx"="Genomics","MS"="Proteomics","Neo"="Mutation")))) %>%filter(Type != "Total")

fig5a$pid <- factor(fig5a$pid)
fig5a$Type <- factor(fig5a$Type,levels = c("Mutation","Fusion","TAA"))
fig5a$Platform <- factor(fig5a$Platform,levels = c("Genomics","Proteomics"))
fig5a$Count <- as.double(fig5a$Count)
fig5a <- fig5a %>% group_by(pid,Type) %>% transmute(Platform, Count,Per=paste0(round(Count[Platform=="Proteomics"] / Count[Platform=="Genomics"] *100),"%")) %>%mutate(Per = replace(Per, Per=="NaN%","")) %>% mutate(Per =replace(Per,Platform=="Proteomics","")) 


theme_set(
  theme_classic()
)

fig5a_p <- fig5a %>% ggplot(aes(x=Type, y=Count,fill=Platform)) + 
  geom_bar(stat="identity",position = "identity") +
  facet_grid(~pid) + 
  labs(title="MB Immunotargetable Antigens", x="", y="Count", fill="Platform") + 
  theme(axis.text.x = element_text(angle =45,hjust = 1 ),plot.title = element_text(size=25,margin=margin(t=20, b=20)))
fig5a_p+geom_text(aes(label=fig5a$Per,group=Type,,vjust = -.5),position = position_dodge((0.8)))




```
### Fig. 6 Immunologic Landscape

### Fig. 6a Group 3 Medulloblastoma Antigen Presentation and Processing Pathway
![Fig. 6a Antigen Presentation and Processing Pathway of MB](/blue/mitchell/changlin/MB_immune/Immune/results/group3_antigenPresentation_2.jpg)

### Fig. 6b Medulloblastoma Immune Deconvolution
*use a 10x 68k pbmc scRNAseq data to deconvolute MB RNAseq/MA data*
```{r message=F, warning=F}
# load("~/changlin/MB_immune/Immune/results/MB_MA_RNAseq_10x_68k_PBMC_Cibersortx.RData")

# Microarray

# fig6b <- draw(hm_mb_ma_68k,annotation_legend_side="bottom",heatmap_legend_side="right",merge_legend=T)

#fig6b
# MB MA scRNAseq 10x 68k sorted pbmc cibersortx 
new_label <- c("Treg","CD4 Memory T Cells","CD8 Naive T Cells", "Dendritic Cells","CD56 NK Cells","CD34 HSC","CD19 B Cells","CD14 Monocytes","CD4 Naive T Cells", "CD4 T Helper Cells")


ma_normal_pheno <- read_tsv("/blue/mitchell/changlin/MB_immune/Immune/meta/Cavalli_pheno.tsv")
ma_pheno <- ma_normal_pheno[c(1:763),]
ma_pheno$subgroup <- factor(ma_pheno$subgroup,levels = col_order)
ma_pheno <- ma_pheno %>% arrange(factor(subgroup,levels = col_order),status,desc(factor(survival)))

mb_ma <- read_tsv("/blue/mitchell/changlin/MB_immune/Immune/results/CIBERSORTx_Job2_Adjusted_MB_MA_10x_68k_pbmc.txt") %>% t() 
colnames(mb_ma) <- as.matrix(mb_ma)[1,]
mb_ma <- mb_ma[c(2:12),]
mb_ma <- mb_ma[-3,]
mb_ma_row <- rownames(mb_ma)
mb_ma <- apply(mb_ma,2,as.numeric)
rownames(mb_ma) <- new_label


hm_mb_ma_68k=Heatmap(pheatmap:::scale_rows(mb_ma),
                      col = colorRamp2(c(-2,0,2),c("blue","white","red")),
                      cluster_columns = F,
                      cluster_rows = T,
                      show_row_names = T,
                      show_column_names = F,
                      name="Gene Expression",
                      column_title = "MB Patients (n=763)",
                      column_title_gp = gpar(fontsize=16,fontface="bold"),
                      row_title = "CibersortX scRNAseq x Microarray",
                      row_title_gp = gpar(fontsize=16,fontface="bold"),
                      border=T,
                      clustering_distance_rows = "euclidean",
                      clustering_method_rows = "complete",
                      heatmap_legend_param = list(
                        title = "Score",
                        title_position = "lefttop-rot",labels_gp=gpar(fontsize=12),title_gp=gpar(fontsize=12)),
                       top_annotation = HeatmapAnnotation(Subtype=ma_pheno$subgroup,
                                             Age=ma_pheno$Age,
                                             Gender=ma_pheno$Gender,
                                             Vital_Status=ma_pheno$status,
                                             Overall_Survival=anno_lines(ma_pheno$survival,gp=gpar(frontsize=8)),
                                               which = "col",
                                             col=list(Subtype=c("WNT"="blue","SHH"="red","Group 3"="yellow","Group 4"="green"),Gender=c("Female"="red","Male"="blue"),Vital_Status=c("Alive"="red","Deceased"="blue"))),
)

draw(hm_mb_ma_68k,annotation_legend_side="bottom",heatmap_legend_side="right",merge_legend=T)
```





